core/controllers/topic_editor.py

```diff
@@ -163,14 +162,17 @@ def get(self, topic_id):
             raise self.PageNotFoundException(
                 Exception('The topic with the given id doesn\'t exist.'))
 
-        skill_ids = topic.get_all_skill_ids()
-
         skill_id_to_description_dict = (
-            skill_services.get_skill_descriptions_by_ids(topic_id, skill_ids))
+            topic_fetchers.get_descriptions_of_linked_skills(topic_id))
+
+        skill_id_to_rubrics_dict = (
+            topic_fetchers.get_rubrics_of_linked_skills(topic_id)
```

Comment:
Why is this in topic_fetchers? This seems to be a skill specific thing, shouldn't it be in skill_services?

Reply:
I think `get_rubrics_of_linked_skills` should be inside `topic_services` and the `get_rubrics_of_linked_skills` should call skill_services (if required).

Through `get_rubrics_of_linked_skills` we are trying to get all the skills linked to a topic and that's more of a topic-specific thing, right?

Reply:
If you look at the get_rubrics_of_linked_skills function, the only 'topic' specific thing it does is topic.get_all_skill_ids. After that everything is skill specific. I feel that should be all in skill services. We can either do what you said (i.e. have topic_fetchers call skill services) or just use skill_services.get_rubrics_of_linked_skills(topic.get_all_skill_ids())

Reply:
Done, added function to skill_services, but calling with only skill_ids.